### ISSUE TITLE
Component type

### ISSUE THREAD

**Issue by sheremet-va** [johnsoncodehk/volar#196]
I have a typing error after the last update inside `component`:

```
Argument of type '{ is: "th" | "td"; role: string; align: string | null; }' is not assignable to parameter of type '{ is: "th" | "td"; } & ("th" | "td")'.ts(2345)
```

My `is` component is a string:
```ts
const el = computed(() => props.header ? 'th' : 'td');
```
```html
<component
    :is="el"
    role="cell"
    :class="$style.tableCell"
    :align="cellAlign"
    :style="cellStyle"
>
```

I know I can use `render` function, but this code works and didn't have any typing errors in the past.

**Comment by johnsoncodehk** [comment]
[v0.25.8](https://github.com/johnsoncodehk/volar/blob/master/CHANGELOG.md#0258) supported `<component :is>` type-checking, but unhandled pass string value to `:is`.

Will fix in next version, thanks to report! You can downgrade to v0.25.7 for now.

**Comment by johnsoncodehk** [comment]
Fixed in 0.25.9. And recommend you update this code to get correct type-checking with `<component>`.

```ts
const el = computed(() => props.header ? 'th' as const : 'td' as const);
```
